src/protocol/decoder.js

```diff
@@ -56,10 +56,34 @@ module.exports = class Decoder {
     return this.canReadBytes(INT64_SIZE)
   }
 
+  // Implementation taken from Node 12
+  // https://github.com/nodejs/node/blob/923d8bc733262cf960b62a02f113cfb0412b5834/lib/internal/buffer.js#L140
+  readBigInt64BE(buffer, offset = 0) {
+    if (isNaN(offset)) return BigInt(0)
+    const first = buffer[offset]
+
+    const last = buffer[offset + 7]
+
+    if (first === undefined || last === undefined) {
+      throw Error(offset, buffer.length - 8)
+    }
+
+    const val =
+      (first << 24) + // Overflow
+      buffer[++offset] * 2 ** 16 +
+      buffer[++offset] * 2 ** 8 +
+      buffer[++offset]
+
+    return (
+      (BigInt(val) << BigInt('32')) +
+      BigInt(
+        buffer[++offset] * 2 ** 24 + buffer[++offset] * 2 ** 16 + buffer[++offset] * 2 ** 8 + last
```

Comment:
Lord almighty. Move this monster out of the return statement. If I'm reading this right, `val` is actually the high bits, so name that `high`, and the enormous line here is the low bits, so name that `low`.